src/common/utils.h

```diff
@@ -88,6 +88,10 @@ auto split(std::string const& s, std::string const& delimiter = " ") -> std::vec
 auto to_lower(std::string const& s) -> std::string;
 auto to_upper(std::string const& s) -> std::string;
 auto trim(const std::string& str, const std::string& whitespace = " \t") -> std::string;
+/// Returns true if the given str matches the given pattern
```

Comment:
I understand that this is not matching the current style of comments in code, but I wanted to bring up if we are thinking of following the javadoc standard used for cpp: https://developer.lsst.io/cpp/api-docs.html.

I think consistency is important, but maybe it's worth breaking away from this to start introducing a bit more verbose documentation. Just thought I would suggest it with this PR.

Reply:
You should open a discussion about it if you trully believe it will be a benefit for the project.

For now though, we use
`// Comment.`

Reply:
You're already explaining how the function works in the function definition, no need to any additional explanation here. (tooltips and other tools will pick up docs from the definition in the cpp file)

Reply:
Will remove comment and stay consistent in the others then.

Reply:
Removed